stripped issuer from label and updated unit tests

shivasheeshyadav opened the pull request:
Fix for issue #551

Issuer is now stripped from the label 

<img src="https://user-images.githubusercontent.com/9421954/82476015-0760d500-9aeb-11ea-9ce3-bd8eb80af4bd.png" width="200">

Tests are updated accordingly.

flocke commented on app/src/main/java/org/shadowice/flocke/andotp/Database/Entry.java:
```diff
@@ -131,6 +131,9 @@ public Entry(String contents) throws Exception {
         String secret = uri.getQueryParameter("secret");
         String label = uri.getPath().substring(1);
+        if(label.contains(":")) {
+            label = label.split(":")[1];
```
I think only splitting the label if it contains a `:` is not the best way of doing this. There could potentially be cases where it contains an `:` without the issuer.
We should probably check if it starts with `<issuer>:` and remove that part specifically.

shivasheeshyadav replied:
@flocke Thanks for the suggestions. I'll make a commit towards this requested change.

SagePtr replied:
Here is how it's done in the Google Authenticator:
https://github.com/google/google-authenticator-android/blob/master/java/com/google/android/apps/authenticator/otp/AccountDb.java#L163

shivasheeshyadav replied:
@flocke @SagePtr
Please review the recent changes

flocke commented on app/src/androidTest/java/org/shadowice/flocke/andotp/ApplicationTest.java:
```diff
@@ -58,12 +60,13 @@
 
-public class ApplicationTest extends ApplicationTestCase<Application> {
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
```
Use

`import static org.junit.Assert.assertEquals;`
`import static org.junit.Assert.assertTrue;`

instead.

flocke commented on app/src/androidTest/java/org/shadowice/flocke/andotp/ApplicationTest.java:
```diff
@@ -23,14 +23,16 @@
-import android.app.Application;
 import android.content.Context;
-import android.test.ApplicationTestCase;
+import androidx.test.InstrumentationRegistry;
```
Deprecated, use `androidx.test.platform.app.InstrumentationRegistry` instead.

flocke commented on app/src/androidTest/java/org/shadowice/flocke/andotp/ApplicationTest.java:
```diff
@@ -168,22 +174,29 @@ public void testEntryURL() throws Exception {
+    @Test
     public void testSettingsHelper() throws KeyStoreException, CertificateException, NoSuchAlgorithmException, IOException {
-        Context context = getContext();
+        Context context = InstrumentationRegistry.getTargetContext();
```
This is deprecated, use `InstrumentationRegistry.getInstrumentation().getTargetContext();` from `androidx.test.platform.app.InstrumentationRegistry` (see comment above).

shivasheeshyadav commented:
> The stripping of the label is now ok, but there are still a lot of problems with the tests. I added some comments how I got them to work on my system (they didn't work with just your PR).

Thanks @flocke 
I've fixed the tests as per your suggestions. Please review.
